code/game/objects/structures/ladders.dm

```diff
@@ -91,6 +91,9 @@
 	if (!is_ghost && !in_range(src, user))
 		return
 
+	if(!up && !down)
+		LateInitialize()
+
```

Comment:
this could be made into a modular edit like

```
/obj/structure/ladder/use(mob/user, is_ghost=FALSE)
	if(!up && !down)
		LateInitialize()
	. = ..()
```

and since we're making a new ladder object for the xenoarch ladder anyways, we could just make the edits we need for it to work directly into the new ladder object